src/tests/test_pubkey.cpp

```diff
@@ -848,12 +848,18 @@ class PK_API_Sign_Test : public Text_Based_Test {
          }
          Test::Result result(test_name.str());
 
-         auto privkey = Botan::create_private_key(algorithm, this->rng(), algo_params, provider);
+         std::unique_ptr<Botan::Private_Key> privkey;
+
+         try {
+            privkey = Botan::create_private_key(algorithm, this->rng(), algo_params, provider);
+         } catch(Botan::Not_Implemented&) {}
```

Comment:
Nit:

```suggestion
         auto privkey = [&]() -> std::unique_ptr<Botan::Private_Key> {
            try {
               return Botan::create_private_key(algorithm, this->rng(), algo_params, provider);
            } catch(Botan::Not_Implemented&) {
               return nullptr
            }
         }();
```

Reply:
Fixed in a different way - ordinarily `create_private_key` returns nullptr if the algorithm is not available/etc (which is maybe not great but that is what it does). So I modified it to first check if we support the named group and if not then just return nullptr same as if ECDSA itself was compiled out.